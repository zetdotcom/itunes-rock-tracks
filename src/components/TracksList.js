import React from 'react';
import { useSelector } from 'react-redux';
import TrackCard from './TrackCard';

import './TrackList.scss';

export default function TracksList() {
	const { tracksList, loading, error } = useSelector(state => state.tracks);
	console.log(error);

	function renderContent() {
		if (error) {
			return <div>Sorry, there was an error</div>;
		}
		if (loading) {
			return <div>Loading ...</div>;
		}
		return (
			<ul className='track-list-grid'>
				{tracksList.map(track => (
					<li className='track-list-grid__item' key={track.trackId}>
						<TrackCard
							id={track.trackId}
							trackName={track.trackName}
							artist={track.artistName}
							price={track.trackPrice}
							artworkUrl={track.artworkUrl100}
						/>
					</li>
				))}
			</ul>
		);
	}

	return <div>{renderContent()}</div>;
}
