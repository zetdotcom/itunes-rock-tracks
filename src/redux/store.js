import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './rootReducer';

const store = state =>
	configureStore({
		reducer: rootReducer,
		preloadedState: state,
	});

export default store;
