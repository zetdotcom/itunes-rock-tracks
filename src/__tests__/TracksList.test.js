import React from 'react';
// import { Provider } from 'react-redux';
import { Provider } from 'react-redux';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import store from '../redux/store';
import TracksList from '../components/TracksList';

// const props = propsOverride => ({
// 	player: {
// 		first_name: 'John',
// 		last_name: 'Doe',
// 		id: 123,
// 		fppg: 100,
// 		images: {
// 			default: {
// 				url: 'testurl',
// 			},
// 		},
// 	},
// 	showFppg: false,
// 	handleCardClick: jest.fn(),
// 	maxFppgId: () => 123,
// 	...propsOverride,
// });

const renderComponent = newStore =>
	render(
		<Provider store={store(newStore)}>
			<TracksList />
		</Provider>
	);

afterEach(cleanup);

describe('TEST PlayerCard', () => {
	test('should match snapshot', () => {
		const { asFragment } = renderComponent();
		expect(asFragment).toMatchSnapshot();
	});

	test('should show error message on error', async () => {
		const newState = {
			tracks: {
				error: 'some error',
			},
		};
		const { getAllByTestId, getByText } = renderComponent(newState);

		const errorMessage = await waitForElement(() => getByText(/ there was an error/i));
		expect(errorMessage).toBeInTheDocument();
	});
	test('should render n tracks', async () => {
		const tracksArr = () =>
			new Array(10)
				.fill()
				.map((x, i) => ({ trackName: 'track name', artist: 'artist', price: '1.00', artworkUrl: 'img', id: i }));
		const mockTracksList = tracksArr();
		const newState = {
			tracks: {
				error: null,
				tracksList: mockTracksList,
			},
		};
		const { getAllByTestId } = renderComponent(newState);
		const tracks = await waitForElement(() => getAllByTestId('track-card'));
		expect(tracks).toHaveLength(10);
	});
});
