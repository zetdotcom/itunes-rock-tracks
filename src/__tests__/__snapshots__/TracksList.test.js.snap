// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST TrackList should match snapshot 1`] = `[Function]`;
